resources-compose/src/androidLibMain/kotlin/KalugaButton.kt

```diff
@@ -0,0 +1,125 @@
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.resources.compose
+
+import android.graphics.Typeface
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.interaction.collectIsPressedAsState
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.ButtonElevation
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.splendo.kaluga.resources.DefaultColors
+import com.splendo.kaluga.resources.StringStyleAttribute
+import com.splendo.kaluga.resources.StyledStringBuilder
+import com.splendo.kaluga.resources.stylable.BackgroundStyle
+import com.splendo.kaluga.resources.stylable.ButtonStateStyle
+import com.splendo.kaluga.resources.stylable.ButtonStyle
+import com.splendo.kaluga.resources.styled
+import com.splendo.kaluga.resources.view.KalugaButton
+import com.splendo.kaluga.resources.view.KalugaLabel
+
+@Composable
+fun KalugaButton.Composable(
+    modifier: Modifier,
+    elevation: ButtonElevation = ButtonDefaults.elevation(),
+    contentPadding: PaddingValues = ButtonDefaults.ContentPadding
+) {
+    val interactionSource: MutableInteractionSource = remember { MutableInteractionSource() }
+    val pressed by interactionSource.collectIsPressedAsState()
+    val stateStyle = style.getStateStyle(isEnabled, pressed)
+    val textStyle = style.getStateTextStyle(isEnabled, pressed)
+    Button(
+        modifier = modifier,
+        enabled = isEnabled,
+        elevation = elevation,
+        colors = ButtonDefaults.buttonColors(backgroundColor = Color.Transparent),
+        shape = stateStyle.backgroundStyle.shape.shape,
+        contentPadding = PaddingValues(0.dp),
+        interactionSource = interactionSource,
+        onClick = action,
+    ) {
+        Box(
+            modifier = Modifier
+                .backgroundStyle(stateStyle.backgroundStyle)
+                .padding(contentPadding)
+                .then(modifier)
+        ) {
+            when (this@Composable) {
+                is KalugaButton.Plain -> KalugaLabel.Plain(text, textStyle)
+                is KalugaButton.Styled -> KalugaLabel.Styled(text)
+                else -> error("unknown button type")
+            }.Composable(modifier = modifier)
+        }
+    }
+}
+
+@Composable
+@Preview
```

Comment:
same point made for BackgroundStyle about `@Preview`
This should be the preferred way to check how our styles look like, NOT running our example apps

Reply:
We can add this as a incremental improvement, resolving for now